File: src/main/java/no/nav/bidrag/beregn/bpsandelunderholdskostnad/beregning/BPsAndelUnderholdskostnadBeregning.java
package no.nav.bidrag.beregn.bpsandelunderholdskostnad.beregning;

import no.nav.bidrag.beregn.bpsandelunderholdskostnad.bo.GrunnlagBeregning;
import no.nav.bidrag.beregn.bpsandelunderholdskostnad.bo.ResultatBeregning;

public interface BPsAndelUnderholdskostnadBeregning {

  ResultatBeregning beregn(GrunnlagBeregning grunnlagBeregning, Boolean beregnMedNyeRegler);

  static BPsAndelUnderholdskostnadBeregning getInstance(){
    return new BPsAndelUnderholdskostnadBeregningImpl();
  }
}

File: src/main/kotlin/no/nav/bidrag/beregn/bpsandelunderholdskostnad/beregning/BPsAndelUnderholdskostnadBeregning.kt
package no.nav.bidrag.beregn.bpsandelunderholdskostnad.beregning

import no.nav.bidrag.beregn.bpsandelunderholdskostnad.bo.GrunnlagBeregning
import no.nav.bidrag.beregn.bpsandelunderholdskostnad.bo.ResultatBeregning

fun interface BPsAndelUnderholdskostnadBeregning {
    fun beregn(grunnlag: GrunnlagBeregning, beregnMedNyeRegler: Boolean): ResultatBeregning

    companion object {
        fun getInstance(): BPsAndelUnderholdskostnadBeregning {
            return BPsAndelUnderholdskostnadBeregningImpl()
        }
    }
}
